services:
  technical_test_database:
    image: postgres:16.1
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    ports:
      - "${POSTGRES_LOCAL_PORT}:${POSTGRES_PORT}"
    restart: always
    volumes:
      - tech-test:/var/lib/postgresql/data
    healthcheck:
      test: "pg_isready --username=${POSTGRES_USER} && psql --username=${POSTGRES_USER} --list"
      timeout: 5s
      retries: 3

  technical_test_api:
    build:
      context: .
      dockerfile: docker/develop/dockerfile

    environment:
      - DATABASE_URL=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@technical_test_database:${POSTGRES_PORT}/${POSTGRES_DB}
      - APP_USERNAME=${APP_USERNAME}
      - APP_EMAIL= ${APP_EMAIL}
      - APP_PASSWORD=${APP_PASSWORD}
      - SECRET_KEY=${SECRET_KEY}

    volumes:
      - ./:/app

    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:${API_CONTAINER_PORT}"

    ports:
      - ${API_PORT}:${API_CONTAINER_PORT}

    depends_on:
        - technical_test_database

volumes:
  tech-test:
    driver: local
